docker.go

```diff
@@ -598,34 +600,46 @@ func (p *DockerProvider) CreateContainer(ctx context.Context, req ContainerReque
 	}
 
 	var tag string
+	customPlatform := req.ImagePlatform
```

Comment:
Super NIT: wdyt about saving this variable and directly use `req.ImagePlatform` elsewhere? Therefore the code itself will be self-contained: I had to double check that the customPlatform variable was in fact initialised with the requested one.